plugins/basic/nexus-repository-maven/src/test/java/org/sonatype/nexus/repository/maven/internal/ArtifactCoordinatesTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import org.sonatype.nexus.repository.maven.internal.ArtifactCoordinates.SignatureType;
+import org.sonatype.nexus.repository.maven.internal.Coordinates.HashType;
+import org.sonatype.nexus.repository.maven.internal.maven2.Maven2PathParser;
+import org.sonatype.sisu.litmus.testsupport.TestSupport;
+
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ArtifactCoordinatesTest
```

Comment:
Try to get into the habbit of always putting class javadocs.  Even in this case of:

```
Tests for {@link ArtifactCoordinates}.
```